test/Microsoft.ML.Tests/Scenarios/Api/Estimators/TrainWithInitialPredictor.cs

```diff
@@ -37,8 +38,10 @@ public void New_TrainWithInitialPredictor()
                 var firstModel = trainer.Fit(trainData);
 
                 // Train the second predictor on the same data.
-                var secondTrainer = new MyAveragedPerceptron(env, new AveragedPerceptronTrainer.Arguments(), "Features", "Label");
-                var finalModel = secondTrainer.Train(trainData, firstModel.Model);
+                var secondTrainer = new AveragedPerceptronTrainer(env, new AveragedPerceptronTrainer.Arguments());
+
+                var trainRoles = new RoleMappedData(trainData, label: "Label", feature: "Features");
```

Comment:
>RoleMappedData [](start = 37, length = 14)

wait, wait, RoleMappedData should not survive :)

Reply:
What is the preferred  way of passing the initialPredictor for online learners if neither Train or Fit takes them in as arguments? Some Transform() ?

---
In reply to: [215821958](https://github.com/dotnet/machinelearning/pull/849#discussion_r215821958) [](ancestors = 215821958,215820969)

Reply:
@Pete Luferenko i added a second TrainContext that contains IDataView for the Training, Validation data, and another Train Method that takes this second TrainContext but that proliferates everywhere in an ugly way. 

I think we should keep the RoleMappedData instead of this spawn, and once it goes away, TrainContext will get cleaned up and so will this example. Objections?

---
In reply to: [215848474](https://github.com/dotnet/machinelearning/pull/849#discussion_r215848474) [](ancestors = 215848474,215821958,215820969)

Reply:
that's ok, let's create an issue for that though.

---
In reply to: [216079699](https://github.com/dotnet/machinelearning/pull/849#discussion_r216079699) [](ancestors = 216079699,215848474,215821958,215820969)